daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1617,28 +1649,9 @@ private[lf] object SBuiltin {
               }
             }: Option[V.ContractId] => (Control, Boolean)
 
-            // TODO (drsk) validate key hash. https://github.com/digital-asset/daml/issues/13897
             machine.disclosureTable.contractIdByKey.get(gkey.hash) match {
               case Some(coid) =>
-                machine.disclosureTable.contractById.get(coid) match {
```

Comment:
As the on ledger contract cache is primed with disclosed contract information (that has been checked), there should no longer be any need to check template types.